commands/root.go

```diff
@@ -36,15 +37,37 @@ func NewRootCmd(name string, isPlugin bool, dockerCli command.Cli) *cobra.Comman
 			if opt.debug {
 				debug.Enable()
 			}
-
 			cmd.SetContext(appcontext.Context())
 			if !isPlugin {
 				return nil
 			}
 			return plugin.PersistentPreRunE(cmd, args)
 		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return cmd.Help()
+			}
+			_ = cmd.Help()
+			return cli.StatusError{
+				StatusCode: 1,
+				Status:     fmt.Sprintf("ERROR: unknown command: %q", args[0]),
+			}
+		},
 	}
-	if !isPlugin {
+	if isPlugin {
```

Comment:
What's the meaning of this logic? Doesn't it run `plugin.PersistentPreRunE` twice now, as `originalPreRun` also seems to contain that call.

Reply:
Ah indeed, I thought it was the other way around in standalone mode but that's not actually!